### PR TITLE
base: rc: nerdctl: Add patch to extend `ps` output

### PR DESCRIPTION
**Pull request by mike-sul**
We need to patch `nerdctl` to get the same info about containers as we get from `docker` CLI command.

Signed-off-by: Mike <mike.sul@foundries.io>

### CODE REVIEW

File: meta-lmp-base/recipes-containers/nerdctl/nerdctl/0001-extend-ps-output.patch [hunk 1]
```diff
@@ -0,0 +1,137 @@
+/iff --git a/src/import/cmd/nerdctl/compose_ps.go b/src/import/cmd/nerdctl/compose_ps.go
```
Comment (quaresmajose) [inline meta-lmp-base/recipes-containers/nerdctl/nerdctl/0001-extend-ps-output.patch:1]: did you test this build? I think there is a typo in `/iff` and bitbake will fail applying the patch

Comment (mike-sul) [reply meta-lmp-base/recipes-containers/nerdctl/nerdctl/0001-extend-ps-output.patch:1]: Yes, I tested it, somehow bitbake doesn't fail,
however it's not correct/typo, i'll fix it.

File: meta-lmp-base/recipes-containers/nerdctl/nerdctl/0001-extend-ps-output.patch [hunk 2]
```diff
@@ -0,0 +1,137 @@
...
++	Labels    map[string]string
+ 	// TODO: "Labels", "LocalVolumes", "Mounts", "Networks", "RunningFor", "Size", "State"
+ }
+ 
+@@ -182,11 +184,13 @@ func printContainers(ctx context.Context, cmd *cobra.Command, containers []conta
+ 			id = id[:12]
+ 		}
+ 
+-		cStatus := formatter.ContainerStatus(ctx, c)
++		cStatus, cState := formatter.ContainerStatus(ctx, c)
```
Comment (doanac) [inline meta-lmp-base/recipes-containers/nerdctl/nerdctl/0001-extend-ps-output.patch:34]: this gets a status and state and then you assign the cStatus to a variable called State. Is there a reason for this like "it make it look more like docker when printed"?

also - maybe just share a before/after example in the commit message.

Comment (mike-sul) [reply meta-lmp-base/recipes-containers/nerdctl/nerdctl/0001-extend-ps-output.patch:34]: >Is there a reason for this like "it make it look more like docker when printed"?

Yes.
We need both `status` (Up 30 hours) and `state` ("running","paused", etc).

Comment (mike-sul) [reply meta-lmp-base/recipes-containers/nerdctl/nerdctl/0001-extend-ps-output.patch:34]: >also - maybe just share a before/after example in the commit message.

That's going to be too huge commit message, I'll just do it in the PR.

Comment (mike-sul) [reply meta-lmp-base/recipes-containers/nerdctl/nerdctl/0001-extend-ps-output.patch:34]: Before
```
{
  "Command": "\"/work/do-it-once.sh\"",
  "CreatedAt": "2022-08-25 12:27:48 +0200 CEST",
  "ID": "2bcbdf6b693c",
  "Image": "hub.foundries.io/msul-dev01/alpine@sha256:aef972662b84a23eb55b87caec80967f2c6a1d6f697cb16822bf75e2bfece82a",
  "Platform": "linux/amd64",
  "Names": "one-shot-failing_one-shot-app_1",
  "Ports": "",
  "Status": "Exited (1) 3 hours ago",
  "Runtime": "io.containerd.runc.v2"
}
{
  "Command": "\"/bin/sh -c python -…\"",
  "CreatedAt": "2022-08-25 12:27:47 +0200 CEST",
  "ID": "39b958b39cf8",
  "Image": "hub.foundries.io/msul-dev01/app-05@sha256:0ec7cd4f8e0443f26f5d1173a6415cb394bebebb9c50339ea1e7a396988c4e63",
  "Platform": "linux/amd64",
  "Names": "app-05_python-www_1",
  "Ports": "0.0.0.0:9985->80/tcp",
  "Status": "Up",
  "Runtime": "io.containerd.runc.v2"
}
{
  "Command": "\"/work/do-it-once.sh\"",
  "CreatedAt": "2022-08-25 12:27:47 +0200 CEST",
  "ID": "db73fc63cc7b",
  "Image": "hub.foundries.io/msul-dev01/alpine@sha256:aef972662b84a23eb55b87caec80967f2c6a1d6f697cb16822bf75e2bfece82a",
  "Platform": "linux/amd64",
  "Names": "one-shot_one-shot-app_1",
  "Ports": "",
  "Status": "Exited (0) 3 hours ago",
  "Runtime": "io.containerd.runc.v2"
}
```
After
```
{
  "Command": "\"/work/do-it-once.sh\"",
  "CreatedAt": "2022-08-25 12:27:48 +0200 CEST",
  "ID": "2bcbdf6b693c",
  "Image": "hub.foundries.io/msul-dev01/alpine@sha256:aef972662b84a23eb55b87caec80967f2c6a1d6f697cb16822bf75e2bfece82a",
  "Platform": "linux/amd64",
  "Names": "one-shot-failing_one-shot-app_1",
  "Ports": "",
  "State": {
    "Status": "stopped",
    "ExitStatus": 1,
    "ExitTime": "2022-08-25T10:27:57.921524509Z"
  },
  "Status": "Exited (1) 3 hours ago",
  "Runtime": "io.containerd.runc.v2",
  "Labels": {
    "com.docker.compose.project": "one-shot-failing",
    "com.docker.compose.service": "one-shot-app",
    "io.compose-spec.config-hash": "8022ec49a307494af0fbffea90ac32e7eb3da4eb894d24b262651a296f31d090",
    "io.containerd.image.config.stop-signal": "SIGTERM",
    "nerdctl/extraHosts": "null",
    "nerdctl/hostname": "one-shot-app",
    "nerdctl/log-uri": "binary:///work/go/src/github.com/containerd/nerdctl/_output/nerdctl?_NERDCTL_INTERNAL_LOGGING=%!F(MISSING)home%!F(MISSING)mike%!F(MISSING).local%!F(MISSING)share%!F(MISSING)nerdctl%!F(MISSING)1935db59",
    "nerdctl/mounts": "[{\"Type\":\"bind\",\"Source\":\"/work/fio/prjs/lmp-device-register/device/compose-apps/one-shot-failing\",\"Destination\":\"/work\",\"Mode\":\"\",\"RW\":false,\"Propagation\":\"\"}]",
    "nerdctl/name": "one-shot-failing_one-shot-app_1",
    "nerdctl/namespace": "default",
    "nerdctl/networks": "[\"one-shot-failing_default\"]",
    "nerdctl/platform": "linux/amd64",
    "nerdctl/state-dir": "/home/mike/.local/share/nerdctl/1935db59/containers/default/2bcbdf6b693caedfdbe73db6eb859a52f6f31f5337fd3da3492b5a5ce9d7a8fc"
  }
}
{
  "Command": "\"/bin/sh -c python -…\"",
  "CreatedAt": "2022-08-25 12:27:47 +0200 CEST",
  "ID": "39b958b39cf8",
  "Image": "hub.foundries.io/msul-dev01/app-05@sha256:0ec7cd4f8e0443f26f5d1173a6415cb394bebebb9c50339ea1e7a396988c4e63",
  "Platform": "linux/amd64",
  "Names": "app-05_python-www_1",
  "Ports": "0.0.0.0:9985->80/tcp",
  "State": {
    "Status": "running",
    "ExitStatus": 0,
    "ExitTime": "0001-01-01T00:00:00Z"
  },
  "Status": "Up",
  "Runtime": "io.containerd.runc.v2",
  "Labels": {
    "com.docker.compose.project": "app-05",
    "com.docker.compose.service": "python-www",
    "io.compose-spec.config-hash": "f1d3106d102f615c90685fe0de7f472e4a7ac402dfe22489250975b42ed0a432",
    "io.containerd.image.config.stop-signal": "SIGTERM",
    "nerdctl/extraHosts": "null",
    "nerdctl/hostname": "python-www",
    "nerdctl/log-uri": "binary:///work/go/src/github.com/containerd/nerdctl/_output/nerdctl?_NERDCTL_INTERNAL_LOGGING=%!F(MISSING)home%!F(MISSING)mike%!F(MISSING).local%!F(MISSING)share%!F(MISSING)nerdctl%!F(MISSING)1935db59",
    "nerdctl/mounts": "[{\"Type\":\"bind\",\"Source\":\"/work/fio/prjs/lmp-device-register/device/compose-apps/app-05\",\"Destination\":\"/var/http\",\"Mode\":\"\",\"RW\":false,\"Propagation\":\"\"}]",
    "nerdctl/name": "app-05_python-www_1",
    "nerdctl/namespace": "default",
    "nerdctl/networks": "[\"app-05_default\"]",
    "nerdctl/platform": "linux/amd64",
    "nerdctl/ports": "[{\"HostPort\":9985,\"ContainerPort\":80,\"Protocol\":\"tcp\",\"HostIP\":\"0.0.0.0\"}]",
    "nerdctl/state-dir": "/home/mike/.local/share/nerdctl/1935db59/containers/default/39b958b39cf80f9167f0cb62bd56bad38071a79c7e7012c165d6a4d10bf2af48"
  }
}
{
  "Command": "\"/work/do-it-once.sh\"",
  "CreatedAt": "2022-08-25 12:27:47 +0200 CEST",
  "ID": "db73fc63cc7b",
  "Image": "hub.foundries.io/msul-dev01/alpine@sha256:aef972662b84a23eb55b87caec80967f2c6a1d6f697cb16822bf75e2bfece82a",
  "Platform": "linux/amd64",
  "Names": "one-shot_one-shot-app_1",
  "Ports": "",
  "State": {
    "Status": "stopped",
    "ExitStatus": 0,
    "ExitTime": "2022-08-25T10:27:57.524302826Z"
  },
  "Status": "Exited (0) 3 hours ago",
  "Runtime": "io.containerd.runc.v2",
  "Labels": {
    "com.docker.compose.project": "one-shot",
    "com.docker.compose.service": "one-shot-app",
    "io.compose-spec.config-hash": "8022ec49a307494af0fbffea90ac32e7eb3da4eb894d24b262651a296f31d090",
    "io.containerd.image.config.stop-signal": "SIGTERM",
    "nerdctl/extraHosts": "null",
    "nerdctl/hostname": "one-shot-app",
    "nerdctl/log-uri": "binary:///work/go/src/github.com/containerd/nerdctl/_output/nerdctl?_NERDCTL_INTERNAL_LOGGING=%!F(MISSING)home%!F(MISSING)mike%!F(MISSING).local%!F(MISSING)share%!F(MISSING)nerdctl%!F(MISSING)1935db59",
    "nerdctl/mounts": "[{\"Type\":\"bind\",\"Source\":\"/work/fio/prjs/lmp-device-register/device/compose-apps/one-shot\",\"Destination\":\"/work\",\"Mode\":\"\",\"RW\":false,\"Propagation\":\"\"}]",
    "nerdctl/name": "one-shot_one-shot-app_1",
    "nerdctl/namespace": "default",
    "nerdctl/networks": "[\"one-shot_default\"]",
    "nerdctl/platform": "linux/amd64",
    "nerdctl/state-dir": "/home/mike/.local/share/nerdctl/1935db59/containers/default/db73fc63cc7b10775b27016cf686c8517dcfa690443de2db593ed5f3d5ed55eb"
  }
}

```

**Comment by ricardosalveti** [comment]
Can you add more details in the commit message and also as part of the patch itself (e.g. if something to upstream, specific to lmp, etc).

File: meta-lmp-base/recipes-containers/nerdctl/nerdctl/0001-extend-ps-output.patch (hunk 2 above)
Comment (doanac) [reply meta-lmp-base/recipes-containers/nerdctl/nerdctl/0001-extend-ps-output.patch:34]: cool. looks like nerdctl is really (ab)using those labels. Doesn't matter here, but I wonder if we should drop the nerdctl/* labels in aklite when reporting state

**Comment by mike-sul** [comment]
> Can you add more details in the commit message and also as part of the patch itself (e.g. if something to upstream, specific to lmp, etc)?

Sure, I'll add details into the commit message.
Not sure it is worth up-streaming at this stage, we just optionally enabling it to do R&D work around it. After that we will know the further steps and whether we want to upstream something or not.

**Comment by mike-sul** [comment]
Closing this PR in favor of https://github.com/foundriesio/meta-lmp/pull/799

**Pull request closed without merging**